sycl/test-e2e/WorkGroupMemory/test-plan.md

```diff
@@ -0,0 +1,148 @@
+# Test plan for [`sycl_ext_oneapi_work_group_memory`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+Functionality provided by the extension is not guarded by any aspects and
+should be supported on every device supported by an implementation.
+
+Therefore, those tests should be launched on every supported device
+configuration that we have.
+
+### Type coverage
+
+New APIs for allocating local memory described by the extension take data type
+as a template argument and therefore tests should be repeated for different
+data types to ensure that everything works correctly.
+
+List of data types which should be covered:
+- some basic C++ data types of different sizes: `char`, `int16_t`, `int`,
+  `double`
+- `half`
+- pointers
+- user-defined types such as `struct`, `union`
+- SYCL-provided data types such as `vec`, `marray`
+- arrays (both bounded and unbounded) of the types listed above
+
+**TODO**: the spec doesn't seem to disallow references and cv-qualified types to
+be used with `work_group_memory` which is likely a bug in the spec.
+
+**TODO**: it would not be possible or reasonable to repeat absolutely every test
+with every data type. Should the specification of which tests should be repeated
+for which data types be a part of the test plan, or should it be an
+implementation detail of the said test plan?
+
+## Tests
```

Comment:
The spec also includes preconditions on the public members stating that they must can be called from only device code, only host code, or both. Should we also have tests that check that the compiler emits some kind of diagnostic upon violation of these conditions?
@gmlueck